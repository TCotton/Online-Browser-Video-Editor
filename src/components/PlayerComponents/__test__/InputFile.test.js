import React from 'react';
import {fireEvent, render, waitFor} from "@testing-library/react";
import FileUpload, {dex} from "../InputFile";

describe('FileUpload', () => {

    const mockDexieRun = jest.fn(n => n);
    const mockDexieRunVideo = jest.fn(n => n);

    beforeEach(() => {
        dex.dexieRun = mockDexieRun;
        dex.dexieRunVideo = mockDexieRunVideo;
    });

    it('should render successfully', () => {
        const {container} = render(<FileUpload />);
        expect(container.querySelector("[data-testid=\"form\"]")).not.toBeNull();
    });

    it('should match snapshot', () => {
        const {baseElement} = render(<FileUpload/>);
        expect(baseElement).toMatchSnapshot();
    });

    it('should have an id and matching label', function () {
        const {container} = render(<FileUpload />);
        expect(container.querySelector("[id=\"file\"]")).not.toBeNull();
        expect(container.querySelector("[for=\"file\"]")).not.toBeNull();
    });

    it('should have an empty error field', function () {
        const {container} = render(<FileUpload />);
        expect(container.querySelector("[class=\"Errors\"]")).not.toBeNull();
        expect(container.querySelector("[class=\"Errors\"]").textContent).toBe('');
    });

    it('should submit correctly',  async () => {
        const video = new File(['(⌐□_□)'], 'video.mp4', { type: 'video/mp4' })
        const {container} = render(<FileUpload />);
        const file = container.querySelector("[data-testid=\"file\"]");

        await waitFor(() => {
            fireEvent.change(file, {
                target: {
                    files: [video]
                }
            });
        })

        expect(mockDexieRun).toHaveBeenCalledTimes(1);
        expect(mockDexieRunVideo).toHaveBeenCalledTimes(1);
    });
});