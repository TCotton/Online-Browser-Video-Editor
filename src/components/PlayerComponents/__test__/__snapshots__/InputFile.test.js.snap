// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload should match snapshot 1`] = `
<body>
  <div>
    <div
      class="app"
    >
      <form
        action="#"
        data-testid="form"
      >
        <div
          class="fileStyling"
        >
          <input
            accept="video/*"
            data-testid="file"
            id="file"
            name="file"
            required=""
            type="file"
            value=""
          />
          <label
            for="file"
          >
            Choose a file
          </label>
        </div>
        <div
          class="Errors"
        />
      </form>
    </div>
  </div>
</body>
`;
